crypto/arm64cpuid.pl

```diff
@@ -104,6 +104,22 @@
 	ret
 .size	_armv8_eor3_probe,.-_armv8_eor3_probe
 
+.globl	_armv8_sve_probe
+.type	_armv8_sve_probe,%function
+_armv8_sve_probe:
+	AARCH64_VALID_CALL_TARGET
+	.long	0x04a03000	// eor z0.d,z0.d,z0.d
```

Comment:
Is `eor` correct here?  If so it would be assigning zero wouldn't it?  Which would be kind of pointless.

Reply:
yes, the only purpose of this instruction is to test if SVE feature is supported by CPU (z0 register only available with SVE). if not, the instruction will cause SIGKILL sent to user process. if yes, it will set a scratch register (z0) to zero with no side-effect to caller.

We have the assembly code internally reviewed in Arm. but not sure if we can find reviewer from other perspective.

Reply:
One thing that could help would be if you (or someone from Arm) were willing to become community maintainer for one of the aarch64 configuration targets that are currently unsupported:
https://www.openssl.org/policies/platformpolicy.html

Reply:
@t8m we are keen to support the community. can you let us know what you expect of the role? in terms of people, equipment, process, effort, etc.

Reply:
As you can see from the description of what the Community support means on the page linked above the requirements are fairly minimal - basically it means the person that is the community contact will need to response to platform-specific defects that are found either by CI or the community in some reasonable time.
Currently the CI is using GitHub Actions their runners which means x86_64 architecture. However we run there CI jobs which use cross-compiler for aarch64 and also run some tests with qemu emulation.

Eventually we would like to be able to run an internal CI based on buildbot and support external buildbot workers which could enable you to use some aarch64 machine(s) hosted by you as CI. But we are not there yet.

Reply:
@t8m thanks for explanation. It is good to know you are running some tests in your CI using qemu emulation.

Internally we run CI on our aarch64 hardware and qemu as well, and as you said, potentially we could help to run standard openssl CI task in the future when it is technically feasible.

We will have some discussion in our team, and see if we can identify person(s) to interface the community. I'll get back to you later. 

Do you have a list of priority of these unsupported aarch64 configurations? (Linux or Android aarch64 could be the priority, I think)

Reply:
It would be really up to you to choose which one you're able to support. And of course you can support also multiple ones.

Reply:
@t8m, could we identify 4 arm engineers contributing to OpenSSL as community maintainers?  We expect there are some backups and supports from different time zone. Besides individual identity is provided, maybe an email alias like openssl-arm-support.arm.com can be set up to reach all of us.  Is there any formal process or guide we can follow to apply for being the community maintainer?

Another question about platform policy. Reading https://www.openssl.org/blog/blog/2021/09/22/OpenSSL3-fips-submission,  there are some platforms that were tested before OpenSSL 3.0 FIPS module was submitted for FIPS validation, are these tested platforms chosen from primary and community platforms?  If not, how a platform can be involved to be tested when validate FIPS module? 

Thanks.

Reply:
You can have multiple people as the contact persons for the platform.

Then one of them simply has to make a PR against the openssl/web repo to make the necessary modifications to the platform policy page and have that PR approved. Once approved and merged its official.

There are no more requirements than what I wrote above in this comment thread.

The FIPS module validation is a separate topic - perhaps @mattcaswell can expand on that.

Reply:
> are these tested platforms chosen from primary and community platforms

Actually no. The FIPS work was sponsored by a number of companies (see https://www.openssl.org/blog/blog/2018/09/25/fips/). We included platforms nominated by our sponsors as well as a couple that the project itself nominated.

Moving forward the OpenSSL Management Committee (OMC) is responsible for maintaining the validation. It is likely that the OMC will want to keep the list of platforms tested to some minimal set. Our primary and secondary lists (currently no platforms are secondary) are likely to influence decisions about any additional platforms to be added if any.

Reply:
Thanks @t8m for the guide, we will make that PR to apply for the community maintainer.   And thanks @mattcaswell for the explanation about FIPS platform.